### PR TITLE
Add initial statements execution to CassandraSession

### PR DESCRIPTION
**Pull request by alar17**
Modify CassandraSession in order to initialize the keyspaces and tables

### CODE REVIEW

File: ts-reaktive-cassandra/src/main/java/com/tradeshift/reaktive/cassandra/CassandraSession.java [hunk 1]
```diff
@@ -14,29 +16,37 @@
...
-
 /**
  * Provides asynchronous, non-blocking access to a cassandra session.
  */
 public class CassandraSession {
     private final akka.persistence.cassandra.session.javadsl.CassandraSession delegate;
-    
-    public CassandraSession(ActorSystem system, String metricsCategory, Function<Session,CompletionStage<Done>> init) {
+
+    public CassandraSession(ActorSystem system, String metricsCategory, Vector<String> statements) {
```
Comment (jypma) [inline ts-reaktive-cassandra/src/main/java/com/tradeshift/reaktive/cassandra/CassandraSession.java]: accept `Seq` (more abstract)

File: ts-reaktive-cassandra/src/main/java/com/tradeshift/reaktive/cassandra/CassandraSession.java [hunk 2]
```diff
@@ -14,29 +16,37 @@
...
+import io.vavr.collection.Vector;
 import io.vavr.control.Option;
-
 /**
  * Provides asynchronous, non-blocking access to a cassandra session.
  */
 public class CassandraSession {
     private final akka.persistence.cassandra.session.javadsl.CassandraSession delegate;
-    
-    public CassandraSession(ActorSystem system, String metricsCategory, Function<Session,CompletionStage<Done>> init) {
```
Comment (jypma) [inline ts-reaktive-cassandra/src/main/java/com/tradeshift/reaktive/cassandra/CassandraSession.java]: could we keep the other constructor around for compatibility?

File: ts-reaktive-cassandra/src/main/java/com/tradeshift/reaktive/cassandra/CassandraSession.java (hunk 1 above)
Comment (jypma) [reply ts-reaktive-cassandra/src/main/java/com/tradeshift/reaktive/cassandra/CassandraSession.java]: call it `initializationStatements`, and give it a bit of java doc.

File: ts-reaktive-cassandra/src/main/java/com/tradeshift/reaktive/cassandra/CassandraSession.java [hunk 3]
```diff
@@ -14,29 +16,37 @@
...
+
+    public CassandraSession(ActorSystem system, String metricsCategory, Vector<String> statements) {
         this.delegate = new akka.persistence.cassandra.session.javadsl.CassandraSession(system,
             new ConfigSessionProvider(system, system.settings().config().getConfig("cassandra-journal")),
             new CassandraSessionSettings(system.settings().config().getConfig("cassandra-journal")),
-            system.dispatcher(), system.log(), metricsCategory, init);
+            system.dispatcher(), system.log(), metricsCategory, s -> createCassandraInit(s, statements));
+    }
+
+    private static CompletionStage<Done> createCassandraInit(Session session, Vector<String> statements) {
```
Comment (vizog) [inline ts-reaktive-cassandra/src/main/java/com/tradeshift/reaktive/cassandra/CassandraSession.java]: I would call it `initializeCassandra` or `executeInitiScripts`

File: ts-reaktive-replication/src/main/java/com/tradeshift/reaktive/replication/VisibilityCassandraSession.java [hunk 4]
```diff
@@ -1,42 +1,37 @@
...
         return keyspace;
     }
-    
-    private static Function<Session,CompletionStage<Done>> createKeyspaceAndTable(Config config) {
+
+    private static Vector<String> initialStatements(Config config) {
         String replStrategy = CassandraPluginConfig.getReplicationStrategy(
             config.getString("replication-strategy"),
             config.getInt("replication-factor"),
             JavaConversions.asScalaBuffer(config.getStringList("data-center-replication-factors")));
```
Comment (vizog) [inline ts-reaktive-replication/src/main/java/com/tradeshift/reaktive/replication/VisibilityCassandraSession.java]: Use `JavaConverters`. `JavaConversions` is deprecated

Comment (alar17) [reply ts-reaktive-replication/src/main/java/com/tradeshift/reaktive/replication/VisibilityCassandraSession.java]: We can update it later as it is not part of this PR and my changes.

Comment (vizog) [reply ts-reaktive-replication/src/main/java/com/tradeshift/reaktive/replication/VisibilityCassandraSession.java]: Created PR #100

**Pull request merged**